### PR TITLE
support reading mzQC

### PR DESCRIPTION
**Pull request by cbielow**

### CODE REVIEW

File: R/io.R [hunk 1]
```diff
@@ -5,10 +5,13 @@
...
 #'
+#' You can in theory also provide any mzQC subelement,
+#' but the resulting mzQC file will not validate since its incomplete.
+#'
 #' The filename should have an '.mzQC' as suffix (warning otherwise).
 #'
-#' @param filepath A filename (with path) to write to.
-#' @param mzqc_obj An mzQC object, which is serialized to JSON and then written to disk
+#' @param filepath A filename (with optional path) to write to.
+#' @param mzqc_obj An MzQCmzQC root object, which is serialized to JSON and then written to disk
```
Comment (mwalzer) [inline R/io.R:14]: So the object notation is a bit confusing. Do you have a quick overview in the doc's somewhere, that explains what's why what?

File: tests/testthat/test-mzQC.R [hunk 2]
```diff
@@ -23,9 +23,18 @@ test_that("isValidMzQC = function(x, ...)", {
 
 
 test_that("fromDatatoMzQC = function(mzqc_class, data)", {
-  data = MzQCcvParameter$new("acc", "myName", "value")
-  data_recovered = fromDatatoMzQC(MzQCcvParameter, list(jsonlite::fromJSON(jsonlite::toJSON(data))))
+  data = rmzqc::MzQCcvParameter$new("acc", "myName", "value")
+  data_recovered = rmzqc::fromDatatoMzQC(rmzqc::MzQCcvParameter, list(jsonlite::fromJSON(jsonlite::toJSON(data))))
```
Comment (mwalzer) [inline tests/testthat/test-mzQC.R:27]: Can you do file tests with testhat?

File: R/mzQC.R [hunk 3]
```diff
@@ -68,39 +68,51 @@ isValidMzQC = function(x, ...)
...
+#' @param data A list of: A datastructure of R lists/arrays as obtained by 'jsonlite::fromJSON()'
+#'
+#' @examples
+#'     data = rmzqc::MzQCcvParameter$new("acc", "myName", "value")
+#'     data_recovered = rmzqc::fromDatatoMzQC(rmzqc::MzQCcvParameter,
+#'                          list(jsonlite::fromJSON(jsonlite::toJSON(data))))
+#'
+#' @export
+#'
+fromDatatoMzQC = function(mzqc_class, data)
```
Comment (mwalzer) [inline R/mzQC.R:108]: Nice, this looks like a really efficient functionality expansion!

**Review by mwalzer — APPROVED** [review]

File: R/io.R (hunk 1 above)
Comment (cbielow) [reply R/io.R:14]: the prefix `MzQC` is used for all datastructures related to the mzQC schema.
Im a bit unhappy with the naming as well.
Recommendations welcome.
I think a common prefix for these objects is a good thing when using an IDE.

File: tests/testthat/test-mzQC.R (hunk 2 above)
Comment (cbielow) [reply tests/testthat/test-mzQC.R:27]: sure. Can add.

**Comment by cbielow** [comment]
added I/O tests as suggested. 
Any more comments?

**Review by mwalzer — APPROVED** [review]

**Pull request merged**